lib/clientBuilder.js

```diff
@@ -244,16 +245,17 @@ proto.fingerprint = function(options) {
     }
     if (msg.type === 'FingerprintServiceReady') {
       if (iframe.contentWindow) {
-        iframe.contentWindow.postMessage(JSON.stringify({
-          type: 'GetFingerprint'
-        }), sdk.options.url);
+        iframe.addEventListener('load', function() {
+          // Make sure the iframe content has loaded before sending the postMessage
+          iframe.contentWindow.postMessage(JSON.stringify({
+            type: 'GetFingerprint'
+          }), sdk.options.url);
+        });
       }
     }
   }
-  oauthUtil.addListener(window, 'message', listener);
-
-  iframe.src = sdk.options.url + '/auth/services/devicefingerprint';
   document.body.appendChild(iframe);
+  oauthUtil.addListener(window, 'message', listener);
```

Comment:
I have the same or similar question: so when we get to this line, the iframe has set its ```src``` attribute and it has been attached to the page. Is there a chance then that we miss the first 'message' if the page loads fast?

In the previous version we had the listener before setting the src and before attaching the iframe to the body of the page.